app/services/course_groups/leadership.rb

```diff
@@ -10,10 +10,8 @@ class Leadership
     delegate :schedules, to: :course_group
 
     def schedule
-      if autumn_schedule_2022?(schedule_date)
-        schedules.find_by!(cohort:, identifier: "npq-leadership-autumn")
-      elsif autumn_schedule_2024?(schedule_date)
-        schedules.find_by!(cohort:, identifier: "npq-leadership-autumn")
+      if autumn_schedule_2024?(schedule_date)
+        schedules.find_by(cohort:, identifier: "npq-leadership-autumn") || schedules.find_by!(cohort:, identifier: "npq-leadership-spring")
```

Comment:
Would this be clearer as a single query?

```suggestion
        schedules.find_by!(cohort:, identifier: %w[npq-leadership-autumn npq-leadership-spring])
```

Reply:
Scratch that actually, your alternative above is more informative.